src/main/java/network/brightspots/rcv/BaseCvrReader.java

```diff
@@ -65,6 +77,30 @@ Map<String, Integer> gatherUnknownCandidates(List<CastVoteRecord> castVoteRecord
       }
     }
 
+    if (includeCandidatesWithZeroVotes) {
+      // Second pass: read all candidates
```

Comment:
would it be helpful to be more explicit about what this second pass is doing. Something like "some providers have explicit candidate lists in their CVR, irrespective of whether those candidates received any votes in the contest. For those providers, this second pass gathers all vendor configured candidates."